### PR TITLE
Topo_id with Namespace & log msg with Logger function 

### PR DESCRIPTION
**Pull request by zzengcs**
The main change in this PR:
1) change all log message with the given Logger function with three message types as Info, Debug, and Error
2) Resolve the PR #153 in this PR for creating/checking/deleting topology id with unique namespace

### CODE REVIEW

File: services/merak-topo/database/database.go [hunk 1]
```diff
@@ -18,10 +18,11 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
-	"log"
 
 	"strings"
 
+	"github.com/futurewei-cloud/merak/services/merak-topo/utils"
```
Comment (r12f) [inline services/merak-topo/database/database.go:24]: `. "github.com/futurewei-cloud/merak/services/merak-topo/utils"`

File: services/merak-topo/database/database.go [hunk 2]
```diff
@@ -40,7 +41,8 @@ func ConnectDatabase() error {
 	})
 
 	if err := client.Ping(Ctx).Err(); err != nil {
-		return fmt.Errorf("ConnectDB: connect DB error %s", err.Error())
+		utils.Logger.Error("ConnectDB: connect DB error %s", err.Error())
```
Comment (r12f) [inline services/merak-topo/database/database.go:44]: can consider to use error.Wrap, just in case of retry, so we can log error as warning in these cases.

Comment (r12f) [reply services/merak-topo/database/database.go:44]: can save `ConnectDB:`.

Comment (r12f) [reply services/merak-topo/database/database.go:44]: log should container parameters of the action.

File: services/merak-topo/database/database.go [hunk 3]
```diff
@@ -50,11 +52,13 @@ func ConnectDatabase() error {
 func SetValue(key string, val interface{}) error {
 	j, err := json.Marshal(val)
```
Comment (r12f) [inline services/merak-topo/database/database.go:53]: would be great to add metrics here, so we can see the call count and latency of this function, same to other apis

Comment (r12f) [reply services/merak-topo/database/database.go:53]: but only metrics, not logs.

Comment (r12f) [reply services/merak-topo/database/database.go:53]:
```
	var err error
	defer metrics.GetMetrics(m, &err)()
```
https://github.com/futurewei-cloud/merak/blob/2d37c70af68c86a3bed8332b70b803ba956f8def/services/merak-agent/evm/alcor.go#L497

**Comment by xieus** [comment]
There is a conflicting file services/merak-topo/handler/topo_deploy.go

File: services/merak-topo/handler/topo_deploy.go [hunk 4]
```diff
@@ -229,10 +228,10 @@ func Topo_deploy(k8client *kubernetes.Clientset, aca_image string, ovs_image str
 							ImagePullPolicy: "IfNotPresent",
 							Command:         []string{"/bin/sh", "-c", "/merak-bin/merak-agent " + aca_parameters},
 							SecurityContext: &sc,
-							Ports: []corev1.ContainerPort{
-								{Name: "gRPC", ContainerPort: 40054},
-								{Name: "prometheus", ContainerPort: 9001},
-							},
+							// Ports: []corev1.ContainerPort{
+							// 	{Name: "gRPC", ContainerPort: 40054},
+							// 	{Name: "prometheus", ContainerPort: 9001},
```
Comment (phudtran) [inline services/merak-topo/handler/topo_deploy.go:233]: I've fixed the deployment issue for vhost in the main branch. Please revert this change and rebase with main.

**Review by phudtran — CHANGES_REQUESTED** [review]

File: services/merak-topo/handler/topo_deploy.go (hunk 4 above)
Comment (zzengcs) [reply services/merak-topo/handler/topo_deploy.go:233]: Thanks for this kind reminder, btw, could you please do notice me when submitting any commits refer to Merak-topo next time? That would make the collaboration much more efficient and effective.

Comment (phudtran) [reply services/merak-topo/handler/topo_deploy.go:233]: I've added you as a code owner for merak-topo awhile back.
https://github.com/futurewei-cloud/merak/blob/main/.github/CODEOWNERS
So you should get notifications for PR review whenever it's modified.
Though it looks like you're missing some permissions on the repo.

Comment (zzengcs) [reply services/merak-topo/handler/topo_deploy.go:233]: In case it happens again, plz ping me a msg then next time.

Comment (phudtran) [reply services/merak-topo/handler/topo_deploy.go:233]: Sure, I can ping you myself if I change something. However, this is not a general solution, as we may get other contributors in the future. The best thing to do would be to fix your permissions such that any contribution to merak-topo will have you auto-requested as a reviewer.

File: services/merak-topo/database/database.go (hunk 2 above)
Comment (zzengcs) [reply services/merak-topo/database/database.go]: addressed, tks

File: services/merak-topo/database/database.go (hunk 3 above)
Comment (zzengcs) [reply services/merak-topo/database/database.go:58]: not use it this time, will try it in future, tks!

File: services/merak-topo/handler/topo_deploy.go (hunk 4 above)
Comment (zzengcs) [reply services/merak-topo/handler/topo_deploy.go]: Thanks! Conflict addressed. All extra kind ping/msg are greatly appreciated for a closer collaboration. 
Current PR includes functions for topo_id with namespace and new logging, free to test with merak-topo:dev, any issue plz free to ping me then.

Comment (xieus) [reply services/merak-topo/handler/topo_deploy.go]: Agreed with @phudtran. Multiple contributors may work for different orgs for the same open-source projects so setting CODEOWNERS would be the solution by default. 

@phudtran which kinds of permission is needed? I saw Zeng has been added as one of contributors.

Comment (zzengcs) [reply services/merak-topo/handler/topo_deploy.go]: @xieus Sure, will follow this way, thanks

Comment (phudtran) [reply services/merak-topo/handler/topo_deploy.go]: > Agreed with @phudtran. Multiple contributors may work for different orgs for the same open-source projects so setting CODEOWNERS would be the solution by default.
> 
> @phudtran which kinds of permission is needed? I saw Zeng has been added as one of contributors.

According to this file both Yan and Zhen are missing write access to the repo
https://github.com/futurewei-cloud/merak/blob/main/.github/CODEOWNERS

**Comment by zzengcs** [comment]
The issues covered in the PR are solved by the PR #178
